src/core/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,301 @@
+      case 'ArrowDown':
+      case 'ArrowRight': {
+        do {
+          newIndex = (newIndex + 1 + labels.length) % labels.length;
+        } while (isTabDisabled(newIndex) && newIndex !== focusedIndex);
+        focusTabAt(newIndex);
+        event.preventDefault();
+        break;
+      }
+      case 'ArrowUp':
+      case 'ArrowLeft': {
+        do {
+          newIndex = (newIndex - 1 + labels.length) % labels.length;
+        } while (isTabDisabled(newIndex) && newIndex !== focusedIndex);
+        focusTabAt(newIndex);
+        event.preventDefault();
+        break;
+      }
+      case 'Enter':
+      case ' ':
+      case 'Spacebar': {
+        event.preventDefault();
+        if (focusActivationMode === 'manual' && focusedIndex !== undefined) {
+          onTabClick(focusedIndex);
+        }
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  return (
+    <div className={cx('iui-tabs-wrapper', `iui-${orientation}`)}>
+      <ul
+        className={cx(
+          'iui-tabs',
+          `iui-${type}`,
+          {
+            'iui-green': color === 'green',
+            'iui-animated': type !== 'default',
+            'iui-large': hasSublabel,
+          },
+          tabsClassName,
+        )}
+        role='tablist'
+        ref={refs}
+        onKeyDown={onKeyDown}
+        {...rest}
+      >
+        {labels.map((label, index) => {
+          const onClick = () => {
+            setFocusedIndex(index);
+            onTabClick(index);
+          };
+          return (
+            <li key={index}>
+              {!React.isValidElement(label) ? (
+                <Tab
+                  label={label}
+                  className={cx({
+                    'iui-active': index === currentActiveIndex,
+                  })}
+                  tabIndex={index === currentActiveIndex ? 0 : -1}
+                  onClick={onClick}
+                  aria-selected={index === currentActiveIndex}
+                />
+              ) : (
+                React.cloneElement(label, {
+                  className: cx(label.props.className, {
+                    'iui-active': index === currentActiveIndex,
+                  }),
+                  'aria-selected': index === currentActiveIndex,
+                  tabIndex: index === currentActiveIndex ? 0 : -1,
+                  onClick: (args: unknown) => {
+                    onClick();
+                    label.props.onClick?.(args);
+                  },
+                })
+              )}
+            </li>
+          );
+        })}
+      </ul>
+      {type !== 'default' && (
+        <div className='iui-tab-stripe' style={stripeStyle} />
+      )}
+      {children && (
+        <div
+          className={cx('iui-tabs-content', contentClassName)}
+          role='tabpanel'
+        >
+          {children}
+        </div>
+      )}
+    </div>
+  );
+};
+
+/**
+ * @deprecated Use `Tabs` instead.
+ */
+export const HorizontalTabs = Tabs;
```

Comment:
@veekeys I know you're not a fan of this approach but I think it's time 😄 
@bentleyvk I heard you're the one responsible for the original name here, so I'm interested in your thoughts too 😄 

---

If we are scared of deprecations, then one alternative would be to make these new `Tabs` components internal, and then expose `HorizontalTabs` and `VerticalTabs` on top of it. Any other way would lead to too much duplicate code.

Reply:
It was one of the first components that I did and I think I went with keeping the same name also iModel.js I think had the same naming.
The bigger question for me is why you named not `Tab` but `HorizontalTab`? 😀 Because I came back from vacation and it was a surprise for me as we already had an item for vertical tabs.
I would go with separate horizontal and vertical tabs comoponents as vertical doesn't support pill style. Also, might help to avoid another breaking change in the future if there will be some bigger differences between these two. But I agree that we could have them as wrappers to `Tabs` component.

Reply:
I vote for separate component.. And we can have any inner code reuse, if needed.

Reply:
Added `VerticalTabs` wrapper. Kept deprecation on individual `HorizontalTab`.